### ISSUE TITLE
Since March 7th update aws_cloudfront_distribution retain_on_delete=true no longer works

### ISSUE THREAD

**Issue by bassmanitram** [terraform-providers/terraform-provider-aws#7869]
### Terraform Version

Terraform v0.11.11

### Affected Resource(s)

* aws_cloudfront_distribution

### Terraform Configuration Files

```hcl
resource "aws_cloudfront_distribution" "core_ui_distribution" {

  #
  # This leaves the distribution disabled rather than deleting it
  # but allows the destroy to work much more quickly
  #
  # An external admin task is required to actually delete the
  # distribution after the destroy has finished.
  #
  # We have to manually remove 
  #    CNAME aliases so they can be used
  #    OAI references so they can be deleted
  #    Certificates so they can be reused
  #    Edge Lambda associations so they can be deleted
  #

  retain_on_delete    = true

  enabled             = true
  is_ipv6_enabled     = true
  default_root_object = "${var.ui_default_document}"
  # ...
```
### Expected Behavior

CloudFront distribution is disabled upon destroy

### Actual Behavior

CloudFront distribution remains enabled

### Steps to Reproduce

1. `terraform apply`
2. `terraform destroy`

**Comment by bflad** [comment]
Hi @bassmanitram 👋 Sorry for the trouble. This change occurred in https://github.com/terraform-providers/terraform-provider-aws/pull/7794/files#diff-a9d90d298f8013ea41caa2d4f3b886a2L809 and was unintentional. Out of curiosity though, is there a specific use case that requires only disabling a distribution? We would expect most environments to not use this flag.

**Comment by bassmanitram** [comment]
Yes, we want the CF distribution "deletion" in TF to be as fast as possible so that it doesn't hang for an interminable time (CF is real bear, as I'm sure you know :) ). Once it's marked disabled, we can do asynchronous stuff to it to disassociate resources and get on with the rest of the state deletion quite quickly. If we don't do this, or state deletion takes a lot longer.

We also noted that CF distribution _creation_ now waits for full deployment whereas it didn't in pre March release. I note, however, that this is actually intentional, but it screws us up.

Is it worth making a feature request for a similar "don't wait for completion" flag on _create_ as retain_on_delete provides for destroy?

**Comment by bflad** [comment]
We'll certainly fix up the regression but I also wonder if there is something else we can do to prevent the need for the flag to begin with. 🤔 

> Is it worth making a feature request for a similar "don't wait for completion" flag on create as retain_on_delete provides for destroy?

Sounds reasonable if that is your desired functionality. 👍

**Comment by bassmanitram** [comment]
> We'll certainly fix up the regression but I also wonder if there is something else we can do to prevent the need for the flag to begin with. thinking

Since CF itself takes ages to do anything, I think you are likely stuck with waiting or disabling - nothing in between would seem to make sense (but, hey, think away :) )

I'll do the feature request in the next couple of days

Thanks

**Comment by bflad** [comment]
Fix submitted with covering acceptance testing to prevent future regressions: #7875

**Comment by nywilken** [comment]
The fix to the `aws_cloudfront_distribution` resource has been merged and will release with version 2.2.0 of the Terraform AWS Provider, likely later today.

**Comment by bflad** [comment]
This has been released in [version 2.2.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#220-march-15-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
